<launch>
  <!-- Arguments -->
  <arg name="cmd_vel_topic" default="/robot/cmd_vel" />
  <arg name="odom_topic" default="odometry/filtered" />
  <arg name="move_forward_only" default="true"/>
  

  <!-- move_base -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <!-- <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" /> -->
    <param name="base_local_planner" value="nav_core_adapter::LocalPlannerAdapter" />
    <param name="LocalPlannerAdapter/planner_name" value="dwb_local_planner::DWBLocalPlanner" />
    <param name="base_global_planner" value="SBPLLatticePlanner" />
    <param name="SBPLLatticePlanner/primitive_filename" value="$(find sbpl_lattice_planner)/matlab/mprim/pr2.mprim"/> 
  <rosparam file="$(find hw_t)/config/sbpl_planner_param.yaml" command="load"/>
    <!-- <param name="base_global_planner"  value="segment_global_planner/SegmentGlobalPlanner"/> -->
    <!-- <rosparam file="$(find segment_global_planner)/param/segment_global_planner_params.yaml" command="load" /> -->
    <rosparam file="$(find hw_t)/config/costmap_common_param.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find hw_t)/config/costmap_common_param.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find hw_t)/config/local_costmap_param.yaml" command="load" />
    <rosparam file="$(find hw_t)/config/global_costmap_param.yaml" command="load" />
    <rosparam file="$(find hw_t)/config/move_base_param.yaml" command="load" />
   <rosparam file="$(find hw_t)/config/dwb_local_planner_params.yaml" command="load" />
    <!-- <rosparam file="$(find segment_global_planner)/param/segment_global_planner_params.yaml" command="load" /> -->
    <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
    <!-- <remap from="odom" to="odometry/filtered" />
    <remap from="map" to="/map"/> -->
    <param name="global_frame" value="map" />
        <param name="robot_base_frame" value="base_link" />
        <param name="odom_frame" value="odom" />
    <!-- <param name="DWAPlannerROS/min_vel_x" value="0.0" if="$(arg move_forward_only)" /> -->
    <param name="TrajectoryPlannerROS/max_vel_theta " value="0.2" />
    <param name="TrajectoryPlannerROS/min_vel_theta " value="0.01" />
    <param name="NavfnROS/use_dijkstra" value="false" />
    <param name="NavfnROS/use_global_planner/GlobalPlanner" value="true" />
      




  </node>
</launch>
